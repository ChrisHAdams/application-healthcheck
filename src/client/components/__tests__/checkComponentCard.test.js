import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import CheckComponentCard from '../checkComponentCard.jsx';

test('Check Component Card Snapshot Test', () => {

  const responseTime = 500;
  const responseCode = 200;

  const component = renderer.create(
    <CheckComponentCard
      name='BBC'
      description='Check the BBC Website'
      type='website'
      expectedResponseTime={responseTime}
      expectedResponseCode={responseCode}
    />,
    null);

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();

});

test('Check Component Card DOM test', () => {
  const responseTime = 500;
  const responseCode = 200;

  const card = shallow(
    <CheckComponentCard
      name='BBC'
      description='Check the BBC Website'
      type='website'
      expectedResponseTime={responseTime}
      expectedResponseCode={responseCode}
    />
  );

  expect(card.find('h3').text()).toEqual('BBC');
  expect(card.find({ id: 'description' }).text()).toEqual('Check the BBC Website');
  expect(card.find({ id: 'type' }).text()).toContain('website');
  expect(card.find({ id: 'expectedResponseTime' }).text()).toContain(responseTime);
  expect(card.find({ id: 'expectedResponseCode' }).text()).toContain(responseCode);

});
