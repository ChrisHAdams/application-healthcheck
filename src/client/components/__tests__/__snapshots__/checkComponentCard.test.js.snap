// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Component Card Snapshot Test 1`] = `
<div
  className="sc-bdVaJa dEiVuS"
>
  <h3>
    BBC
  </h3>
  <h4
    id="description"
  >
    Check the BBC Website
  </h4>
  <p
    id="type"
  >
    Check Type: 
    website
  </p>
  <p
    id="expectedResponseCode"
  >
    Expected Response Code: 
    200
  </p>
  <p
    id="actualResponseCode"
  >
    Actual Response Code: 
  </p>
  <p
    id="expectedResponseTime"
  >
    Expected Response Time: 
    500
    ms
  </p>
  <p
    id="actualResponseTime"
  >
    Actual Response Time: 
  </p>
</div>
`;
